### PR TITLE
CoreNLP Task Impl.

### PR DESCRIPTION
**Pull request by KaisongHuang**

### CODE REVIEW

File: libs/corenlp/ner.py [hunk 1]
```diff
@@ -3,8 +3,47 @@
...
 
-    def __init__(self, config):
-        print(config)
+    def __init__(self, spark_context):
+        self.sc = spark_context
+
+        # Get an instance of the Properties
+        self.props = self.sc._jvm.java.util.Properties()
+        self.setProperty("annotators", "tokenize,ssplit,pos,lemma,ner")
+        self.setProperty("ner.useSUTime", "false");
```
Comment (r-clancy) [inline libs/corenlp/ner.py]: This makes it faster, but we lose tagging of time (i.e., "next wednesday", "3pm", etc.) - might want to change this in the future.

File: libs/corenlp/ner.py [hunk 2]
```diff
@@ -3,8 +3,47 @@
...
+        words = 0
+
+        # For each parsed doc...
+        for paragraph in data:
+
+            # Keep track of sentences in the paragraph
+            par = []
+
+            # For each sentence in the paragraph, fetch the entities
+            anno = self.sc._jvm.edu.stanford.nlp.pipeline.Annotation(paragraph)
```
Comment (r-clancy) [inline libs/corenlp/ner.py]: I think we want to run NER over the entire document once (saves overhead of creating multiple `CoreDocument`) and iterate over the sentences to extract entities... see [here](https://github.com/ljj7975/solr-evaluation/blob/master/src/main/scala/ca/uwaterloo/SIGIR/NamedEntityRecognition.scala#L39).

No need to create the `Annotation` and `SentenceAnnotation`... the pipeline should already do this stuff.

File: libs/corenlp/ner.py (hunk 1 above)
Comment (KaisongHuang) [reply libs/corenlp/ner.py]: I'll find which jar is missing here if we need this property enabled later.

**Comment by KaisongHuang** [comment]
Refactor the codes.

**Pull request closed without merging**